web/modules/custom/dbcdk_community/src/Profile/ProfileRepository.php

```diff
@@ -0,0 +1,357 @@
+<?php
+/**
+ * @file
+ * Class definition for ProfileRepository.
+ */
+
+namespace Drupal\dbcdk_community\Profile;
+
+
+use DateTime;
+use DBCDK\CommunityServices\Api\CommunityRoleApi;
+use DBCDK\CommunityServices\Api\ProfileApi;
+use DBCDK\CommunityServices\Api\QuarantineApi;
+use DBCDK\CommunityServices\Model\CommunityRole;
+use DBCDK\CommunityServices\Model\Profile as ModelProfile;
+use DBCDK\CommunityServices\Model\Quarantine;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+
+/**
+ * A repository for retrieving profiles with related objects.
+ *
+ * Related objects can be quarantines and community roles.
+ */
+class ProfileRepository {
+  use LoggerAwareTrait;
+
+  /**
+   * The API to use when retrieving profiles.
+   *
+   * @var \DBCDK\CommunityServices\Api\ProfileApi $profileApi
+   */
+  protected $profileApi;
+
+  /**
+   * The API to use when retrieving quarantines.
+   *
+   * @var \DBCDK\CommunityServices\Api\QuarantineApi $quarantineApi
+   */
+  protected $quarantineApi;
+
+  /**
+   * The API to use when retrieving community roles.
+   *
+   * @var \DBCDK\CommunityServices\Api\CommunityRoleApi
+   */
+  protected $communityRoleApi;
+
+  /**
+   * ProfileRepository constructor.
+   *
+   * @param \DBCDK\CommunityServices\Api\ProfileApi $profile_api
+   *   The profile api to use.
+   * @param \DBCDK\CommunityServices\Api\QuarantineApi $quarantine_api
+   *   The quarantine api to use.
+   * @param \DBCDK\CommunityServices\Api\CommunityRoleApi $community_role_api
+   *   The community role api to use.
+   */
+  public function __construct(ProfileApi $profile_api, QuarantineApi $quarantine_api, CommunityRoleApi $community_role_api) {
+    $this->profileApi = $profile_api;
+    $this->quarantineApi = $quarantine_api;
+    $this->communityRoleApi = $community_role_api;
+    $this->logger = new NullLogger();
+  }
+
+  /**
+   * Transform a filter data structure to the format expected by APIs.
+   *
+   * @param array $filter
+   *   The filter. This can contain entries such as where, limit, offset.
+   *
+   * @return string
+   *   A JSON encoded version of the filter.
+   *
+   * @see https://docs.strongloop.com/display/public/LB/Querying+data
+   */
+  protected function processFilter(array $filter) {
+    return (!empty($filter)) ? json_encode($filter) : NULL;
```

Comment:
According to the DocBlock, this should return a string. Either the return should be changed or the DocBlock needs an update :relaxed: